### PR TITLE
[Feature] Add multi behavior dataset

### LINKED ISSUE #42: [💡SUG] Multi behavior dataset usage
Hi, thanks for your hard work! The repo is quite useful and inspiring.
As there're many multi-behavior recommendation models based on GNN, I wonder is there any plan to support multi-behavior dataset loading in this repo and how to use it?
Thanks for your quick reply.

### PR DESCRIPTION
**Pull request by Tokkiu**
Fix #42 @hyp1231 
Add implementation of the multi-behavior dataset.
Plz check it if any could be improved

### CODE REVIEW

File: recbole_gnn/data/dataset.py [hunk 1]
```diff
@@ -101,6 +101,54 @@ def build(self):
             dataset.session_graph_construction()
         return datasets
 
+class MultiBehaviorDataset(SessionGraphDataset):
+
+    def session_graph_construction(self):
+        self.logger.info('Constructing Multi-behavior session graphs.')
```
Comment (hyp1231) [inline recbole_gnn/data/dataset.py:107]:
```suggestion
        self.logger.info('Constructing multi-behavior session graphs.')
```

File: recbole_gnn/data/dataset.py [hunk 2]
```diff
@@ -101,6 +101,54 @@ def build(self):
...
+        self.item_behavior_list_field = self.config['ITEM_BEHAVIOR_LIST_FIELD']
+        item_seq = self.inter_feat[self.item_id_list_field]
+        item_seq_len = self.inter_feat[self.item_list_length_field]
+        if self.item_behavior_list_field == None:
+            # To be compatible with existing datasets
+            item_behavior_seq = torch.tensor([0] * len(item_seq))
+            self.field2id_token['behavior_id'] = {0:'interaction'}
```
Comment (hyp1231) [inline recbole_gnn/data/dataset.py:114]:
```suggestion
            self.behavior_id_field = config['Behavior_ID_FIELD']
            self.field2id_token[self.behavior_id_field] = {0: 'interaction'}
```

Maybe the field name of behavior ID can also be customized? 
```yaml
Behavior_ID_FIELD: behavior_id
```
or fixed as `behavior_id` is also ok. Just a suggestion.

File: recbole_gnn/data/dataset.py [hunk 3]
```diff
@@ -101,6 +101,54 @@ def build(self):
...
+            seq, idx = torch.unique(seq, return_inverse=True)
+            alias_seq = idx.squeeze(0)[:item_seq_len[i]]
+            alias_inputs.append(alias_seq)
+            # No repeat click
+            edge = torch.stack([alias_seq[:-1], alias_seq[1:]]).unique(dim=-1)
+            edge_index.append(edge)
+
+        nx = {}
+        for k, v in x.items():
+            behavior_name = self.id2token('behavior_id', k)
```
Comment (hyp1231) [inline recbole_gnn/data/dataset.py:143]: The same as above, about whether the field name `behavior_id` can be customized.

**Comment by Tokkiu** [comment]
@hyp1231 Thanks for your kind comments! It makes sense and I have fixed them all while testing to construct a multi-behavior dataset.
Plz, check the newest version.

**Review by hyp1231 — COMMENTED** [review]
Thanks for your quick reply! I still have one concern which is listed above.

**Review by hyp1231 — APPROVED** [review]
Excellent! Thanks for the great efforts. :)

**Pull request merged**